File: tools/sdks/go/examples/quickstart_get.go
//go:build ignore

// quickstart_get — Fetch the authenticated principal via GET /v1/users/me.
//
// Customer concept: canonical "who am I" probe. Verify PAT validity,
// scopes, and tenant binding before issuing follow-up calls.
//
// Run: CORELINK_PAT=$PAT go run examples/quickstart_get.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
	"time"
)

func main() {
	api := os.Getenv("CORELINK_API_URL")
	if api == "" {
		api = "https://sandbox.corelink.dev"
	}
	pat := os.Getenv("CORELINK_PAT")
	if pat == "" {
		fmt.Fprintln(os.Stderr, "error: CORELINK_PAT env var is required")
		os.Exit(2)
	}
	req, err := http.NewRequest("GET", api+"/v1/users/me", nil)
	if err != nil {
		fmt.Fprintf(os.Stderr, "build request: %v\n", err)
		os.Exit(1)
	}
	req.Header.Set("Authorization", "Bearer "+pat)
	req.Header.Set("Accept", "application/json")
	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Do(req)
	if err != nil {
		fmt.Fprintf(os.Stderr, "http: %v\n", err)
		os.Exit(1)
	}
	defer resp.Body.Close()
	raw, _ := io.ReadAll(resp.Body)
	if resp.StatusCode/100 != 2 {
		fmt.Fprintf(os.Stderr, "server returned %d: %s\n", resp.StatusCode, string(raw))
		os.Exit(1)
	}
	var pretty bytes.Buffer
	if err := json.Indent(&pretty, raw, "", "  "); err != nil {
		pretty.Write(raw)
	}
	fmt.Printf("status=%d\n%s\n", resp.StatusCode, pretty.String())
}

File: tools/sdks/python/examples/quickstart_get.py
#!/usr/bin/env python3
"""quickstart_get — Fetch the authenticated principal via GET /v1/users/me.

Customer concept: the canonical "who am I" probe. Use it to verify PAT
validity, scopes, and tenant binding before issuing follow-up calls.

Run:
    CORELINK_PAT=$PAT python3 examples/quickstart_get.py
"""

from __future__ import annotations

import json
import os
import sys
import urllib.error
import urllib.request


def main() -> int:
    api = os.environ.get("CORELINK_API_URL", "https://corelink-api.humangr.com")
    pat = os.environ.get("CORELINK_PAT")
    if not pat:
        print("error: CORELINK_PAT env var is required", file=sys.stderr)
        return 2

    req = urllib.request.Request(
        url=f"{api}/v1/users/me",
        method="GET",
        headers={"Authorization": f"Bearer {pat}", "Accept": "application/json"},
    )
    try:
        with urllib.request.urlopen(req, timeout=10) as resp:
            payload = json.loads(resp.read().decode("utf-8"))
            print(f"status={resp.status}")
            print(json.dumps(payload, indent=2))
    except urllib.error.HTTPError as exc:
        print(f"http_error status={exc.code} body={exc.read().decode('utf-8', 'replace')}", file=sys.stderr)
        return 1
    except urllib.error.URLError as exc:
        print(f"network_error: {exc.reason}", file=sys.stderr)
        return 1
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
